src/scib_metrics/_pcr.py

```diff
@@ -0,0 +1,80 @@
+from typing import Optional
+
+import jax
+import jax.numpy as jnp
+
+from scib_metrics.utils import one_hot, pca
+
+from ._types import NdArray
+
+
+def principal_component_regression(
+    X: NdArray,
+    covariate: NdArray,
+    categorical: Optional[bool] = False,
```

Comment:
nit: Not optional if default is False (it's just an alias for Union[bool, None])